extensions/hdfs-storage/src/main/java/io/druid/storage/hdfs/HdfsDataSegmentPusher.java

```diff
@@ -93,10 +93,8 @@ private DataSegment createDescriptorFile(DataSegment segment, Path outDir, final
   {
     final Path descriptorFile = new Path(outDir, "descriptor.json");
     log.info("Creating descriptor file at[%s]", descriptorFile);
-    ByteStreams.copy(
-        ByteStreams.newInputStreamSupplier(jsonMapper.writeValueAsBytes(segment)),
-        new HdfsOutputStreamSupplier(fs, descriptorFile)
-    );
+    final ByteSource byteSource = ByteSource.wrap(jsonMapper.writeValueAsBytes(segment));
```

Comment:
This is not a requirement, but you can simply chain the methods here to make it more readable

```java
ByteSource
    .wrap(jsonMapper.writeValueAsBytes(segment))
    .copyTo(new HdfsOutputStreamSupplier(fs, descriptorFile));
```